File: clean-care-frontend/src/types/conversation.ts
export interface Conversation {
  conversation_id: string;
  title: string;
  status: string;
  created_at: string;
  last_message_at: string;
}

export interface Message {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  trace_id?: string;
  created_at?: string;
}

File: internal/model/conversation.go
package model

import "time"

type Conversation struct {
	ID            string    `json:"conversation_id"`
	UserID        string    `json:"-"`
	Title         string    `json:"title"`
	Status        string    `json:"status"`
	CreatedAt     time.Time `json:"created_at"`
	LastMessageAt time.Time `json:"last_message_at"`
}

type Message struct {
	ID              string    `json:"message_id"`
	ConversationID  string    `json:"conversation_id,omitempty"`
	Role            string    `json:"role"`
	Content         string    `json:"content"`
	TraceID         string    `json:"trace_id,omitempty"`
	ClientMessageID string    `json:"client_message_id,omitempty"`
	CreatedAt       time.Time `json:"created_at"`
}
